Fix macOS Runner when Meson is Already Installed, and Add macOS 15

dgreatwood opened the pull request:
It appears that the github macOS runners now have meson preinstalled.

This was causing an issue when our macos.yaml did:
`brew install meson`
We now have macos.yaml check for meson, and install it only if meson is not already present.

Also, added the newly released macOS-15 as one of the OS to test in macos.yaml.

kiplingw commented:
Looks fine to me @dgreatwood. Good work. If nothing (new) breaks in CI, I'll merge after its done.

dgreatwood commented:
Looks like macOS-12, 14 and 15 are OK, but still a problem with 13. I'll
take another look.

On Sun, Oct 13, 2024 at 3:28 PM Kip ***@***.***> wrote:

> Looks fine to me @dgreatwood <https://github.com/dgreatwood>. Good work.
> If nothing (new) breaks in CI, I'll merge after its done.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pistacheio/pistache/pull/1253#issuecomment-2409210548>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFMA227CHTTY73DGI67VX3Z3LXXXAVCNFSM6AAAAABP32566SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBZGIYTANJUHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
NOTICE: This email and its attachments may contain privileged and 
confidential information, only for the viewing and use of the intended 
recipient. If you are not the intended recipient, you are hereby notified 
that any disclosure, copying, distribution, acting upon, or use of the 
information contained in this email and its attachments is strictly 
prohibited and that this email and its attachments must be immediately 
returned to the sender and deleted from your system. If you received this 
email erroneously, please notify the sender immediately.  Xage Security, 
Inc. and its affiliates will never request personal information (e.g., 
passwords, Social Security numbers) via email.  Report suspicious emails to 
***@***.*** ***@***.***>

dgreatwood commented:
Fixed now for macOS-13 also; branch updated.

On Sun, Oct 13, 2024 at 3:45 PM Duncan Greatwood ***@***.***>
wrote:

> Looks like macOS-12, 14 and 15 are OK, but still a problem with 13. I'll
> take another look.
>
> On Sun, Oct 13, 2024 at 3:28 PM Kip ***@***.***> wrote:
>
>> Looks fine to me @dgreatwood <https://github.com/dgreatwood>. Good work.
>> If nothing (new) breaks in CI, I'll merge after its done.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/pistacheio/pistache/pull/1253#issuecomment-2409210548>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAFMA227CHTTY73DGI67VX3Z3LXXXAVCNFSM6AAAAABP32566SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBZGIYTANJUHA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

-- 
NOTICE: This email and its attachments may contain privileged and 
confidential information, only for the viewing and use of the intended 
recipient. If you are not the intended recipient, you are hereby notified 
that any disclosure, copying, distribution, acting upon, or use of the 
information contained in this email and its attachments is strictly 
prohibited and that this email and its attachments must be immediately 
returned to the sender and deleted from your system. If you received this 
email erroneously, please notify the sender immediately.  Xage Security, 
Inc. and its affiliates will never request personal information (e.g., 
passwords, Social Security numbers) via email.  Report suspicious emails to 
***@***.*** ***@***.***>
